internal/cron/verify_domain_ownership.go

```diff
@@ -0,0 +1,88 @@
+package cron
+
+import (
+	"context"
+	"errors"
+	"net"
+
+	"github.com/oursky/pageship/internal/db"
+	"github.com/oursky/pageship/internal/models"
+	"github.com/oursky/pageship/internal/time"
+	"go.uber.org/zap"
+)
+
+type VerifyDomainOwnership struct {
+	Schedule                     string
+	DB                           db.DB
+	MaxConsumeActiveDomainCount  uint
+	MaxConsumePendingDomainCount uint
+}
+
+func (v *VerifyDomainOwnership) Name() string { return "verify-domain-ownership" }
+
+func (v *VerifyDomainOwnership) CronSchedule() string { return v.Schedule }
+
+func (v *VerifyDomainOwnership) Run(ctx context.Context, logger *zap.Logger) error {
+	now := time.SystemClock.Now().UTC()
+
+	return db.WithTx(ctx, v.DB, func(c db.Tx) error {
+		verified := true
+		unverified := false
+		domainVerifications := []*models.DomainVerification{}
+		verifiedRecords, _ := c.ListDomainVerifications(ctx, nil, &v.MaxConsumeActiveDomainCount, &verified)
+		unverifedRecords, _ := c.ListDomainVerifications(ctx, nil, &v.MaxConsumePendingDomainCount, &unverified)
+		for _, verifiedRecord := range verifiedRecords {
+			domainVerifications = append(domainVerifications, verifiedRecord)
+		}
+		for _, unverifiedRecord := range unverifedRecords {
+			domainVerifications = append(domainVerifications, unverifiedRecord)
+		}
+
+		var consumedCount = 0
+		for _, domainVerification := range domainVerifications {
+			key, value := domainVerification.GetTxtRecord()
+			values, err := net.LookupTXT(key)
```

Comment:
@kiootic plan to add test case for this worker. May i know if you have any prefer way other than patched with https://pkg.go.dev/github.com/foxcpp/go-mockdns#section-readme to mock out this line for testing ?

Reply:
I would suggest making a interface for this, just like the `Clock` interface.

```go
type DNSLookuper interface {
  LookupTXT(name string) ([]string, error)
}


var SystemDNS DNSLookuper = systemDNS{}

type systemDNS struct {}

func (s systemDNS) LookupTXT(name string) ([]string, error) {
  return net.LookupTXT(name)
}


type MockDNS struct {
    TXTRecords map[string][]string
}

func (s *MockDNS) LookupTXT(name string) ([]string, error) {
  /* ... */
}